python/lsst/obs/base/cameraMapper.py

```diff
@@ -1106,6 +1106,14 @@ def _getBestFilter(self, storedLabel, idFilter):
                 newLabel = list(definitions)[0].makeFilterLabel()
                 return newLabel
             elif definitions:
+                # Could this be a band name that only ever identifies one of a
+                # series of filters?
+                nonAfw = {f for f in definitions if f.afw_name is None}
+                if len(nonAfw) == 1:
```

Comment:
I am surprised that this block is actually necessary, since storedLabel here should have a valid physical label, which means the old code below should "just work" for CFHT (`len(bands) == 1` should be true, since all of those different physical labels have the same band in CFHT).

Reply:
It's necessary because there are lots of places where you don't have a physical filter (e.g., your last example on Jira, and more generally any pre-`FilterLabel` calexp).